### PR TITLE
Spec

### PR DESCRIPTION
**Pull request by pinglinh**
Please leave a comment with clues on how to do a test for this type of code.

### CODE REVIEW

File: lib/echo_server.rb [hunk 1]
```diff
@@ -1,53 +1,45 @@
...
-      @view.print_user_input(@user_input)
-      @user_input = STDIN.readline.chomp
+class UserEchoServer
+  def initialize(input=STDIN, output=STDOUT)
+    @input = input
+    @output = output
+    @view = View.new(@output)
+  end
+
+  def echo_server_start
```
Comment (cnak) [inline lib/echo_server.rb]: Rename this to start

File: lib/echo_server.rb [hunk 2]
```diff
@@ -1,53 +1,45 @@
...
   end
 
   def exit_entered?
     @user_input == "exit"
   end
 
+  def readline
+    @input.readline.chomp
+  rescue EOFError
+    "exit"
```
Comment (cnak) [inline lib/echo_server.rb]: Perhaps change "exit" to a constant perhaps

File: lib/command_line_view.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# class CommandLineView
```
Comment (cnak) [inline lib/command_line_view.rb:1]: Remove commented out code

File: lib/command_line_view.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+
+
+class CommandLineView
+  def initialize(output=STDOUT)
+    @output = output
+  end
+
+  def welcome_message
+    @output.print "Welcome!\n"
```
Comment (pinglinh) [inline lib/command_line_view.rb:33]: Change all to puts or include \n in the echo_user_line method

File: lib/user_echo_server.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+require_relative 'user_echo_input'
+
+class UserEchoServer
+  EXIT = "exit"
+
+  def initialize(input=STDIN, output=STDOUT)
+    @input = input
+    @output = output
+    @command_line_view = CommandLineView.new(@output)
+    @user_echo_input = UserEchoInput.new(@input)
```
Comment (cnak) [inline lib/user_echo_server.rb:11]: Use dependency injection. No need tobe explicit.

File: lib/user_echo_server.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
+require_relative 'command_line_view'
+require_relative 'user_echo_input'
+
+class UserEchoServer
+  EXIT = "exit"
+
+  def initialize(input=STDIN, output=STDOUT)
+    @input = input
+    @output = output
```
Comment (pinglinh) [inline lib/user_echo_server.rb:9]: @output = CommandLineView.new

File: lib/user_echo_server.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
+require_relative 'command_line_view'
+require_relative 'user_echo_input'
+
+class UserEchoServer
+  EXIT = "exit"
+
+  def initialize(input=STDIN, output=STDOUT)
+    @input = input
```
Comment (pinglinh) [inline lib/user_echo_server.rb:8]: @input = UserEchoInput.new

File: lib/user_echo_server.rb [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  #   @user_input = readline
+  #   while exit_entered? == true
+  #     @command_line_view.goodbye_message
+  #   end
+  #   @output.puts @user_input
+  #   @user_input = readline
+  # end
+
+  server = UserEchoServer.new
+  server.start
```
Comment (cnak) [inline lib/user_echo_server.rb:51]: Remove hanging code

File: lib/user_echo_server.rb [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+  def readline
+    @input.readline.chomp
+  rescue EOFError
+    EXIT
+  end
+end
+
+
+  # def start
```
Comment (cnak) [inline lib/user_echo_server.rb:39]: Remove commented out code

**Pull request merged**